worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/factory/parser/RichParser.java

```diff
@@ -123,7 +123,21 @@ public E parseFromInput(String input, ParserContext context) throws InputParseEx
      * @param index         the index of the argument to get suggestions for.
      * @return a stream of suggestions matching the given input for the argument at the given index.
      */
-    protected abstract Stream<String> getSuggestions(String argumentInput, int index);
+    protected Stream<String> getSuggestions(String argumentInput, int index) {
+        return Stream.empty();
```

Comment:
This should default to the new method with an empty parser context

Reply:
> This should default to the new method with an empty parser context

I've swapped them round https://github.com/IntellectualSites/FastAsyncWorldEdit/pull/2613/commits/5c257f60ea9ec002e407b2a4a8a7c6ede6bee5e7

Reply:
I meant that they should both default to each other so parsers can implement one of either method and it will still work. Similar is already done in extents https://github.com/IntellectualSites/FastAsyncWorldEdit/blob/main/worldedit-core/src/main/java/com/sk89q/worldedit/extent/InputExtent.java#L52-L59

Reply:
Maybe a dumb question, but would there be any risk if someone did not overwrite one or both of them when using the API? Would the getSuggestions not just loop?

Reply:
Yeah I suppose we're not necessarily going to always do suggestions? Return it to how it was but add a deprecation to it I think, as that method should never be called; overriding is fine but calling is not

Reply:
Added the Deprecated tag & a note: https://github.com/IntellectualSites/FastAsyncWorldEdit/pull/2613/commits/2873359ec458a8b7d3fe8566fc4d01a826c66a55